clients/imodels-client-management/src/base/internal/AxiosRestClient.ts

```diff
@@ -82,3 +98,34 @@ export class AxiosRestClient implements RestClient {
     }
   }
 }
+
+/** Default implementation for {@link HttpResponseHeadersFactory}. */
+export class AxiosHeadersAdapterFactory implements HttpResponseHeadersFactory {
```

Comment:
should be in a separate file

Reply:
It's a pretty simple class, which is only used in this file. Wouldn't separating it to another file make it harder to read?

Reply:
currently it's a bit hidden under `AxiosRestClient`, maybe `AxiosHeadersAdapterFactory` and `AxiosHeadersAdapter` could be both moved into a separate file together.